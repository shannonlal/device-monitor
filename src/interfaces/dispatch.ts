export interface IDispatch {
    dispatch?: (action: any) => Promise<any>;
}
