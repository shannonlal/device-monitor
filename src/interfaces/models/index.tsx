export interface IDeviceDetailsModel {
    name?: string;
    diskFree?: number;
    appVersion: string;
    appBuild: string;
    operatingSystem: string;
    osVersion: string;
    platform: 'ios' | 'android' | 'electron' | 'web';
    memUsed?: number;
    diskTotal?: number;
    model: string;
    manufacturer: string;
    uuid: string;
    isVirtual: boolean;
}

export interface IUser {
    eMail: string;
    firstName: string;
    lastName: string;
}

export interface IAuthorization {
    eMail: string;
    password: string;
}

export interface IWifiSummary {
    signalStrength: number;
    ssid: string;
}

// This wifi details interface will expand in the future.  Stubbed for now
export interface IWifiDetails {
    signalStrength: number;
    ssid: string;
    security: string;
}
