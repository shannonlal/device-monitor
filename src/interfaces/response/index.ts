export interface IResponse<T> {
    content?: T;
    httpStatus: number;
    errorMessage?: string;
}
