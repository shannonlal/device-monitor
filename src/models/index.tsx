export type DeviceDetailsModel = {
    name?: string;
    diskFree?: number;
    appVersion: string;
    appBuild: string;
    operatingSystem: string;
    osVersion: string;
    platform: 'ios' | 'android' | 'electron' | 'web';
    memUsed?: number;
    diskTotal?: number;
    model: string;
    manufacturer: string;
    uuid: string;
    isVirtual: boolean;
};
