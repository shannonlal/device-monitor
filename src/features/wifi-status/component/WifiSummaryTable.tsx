import React from 'react';
import { IonText, IonRow, IonCol, IonGrid } from '@ionic/react';
import { IWifiSummary } from '../../../interfaces/models';
import DeviceField from '../../../components/common/DeviceField';

type WifiSummaryTableProps = {
    ssidLabel: string;
    signalStength: string;
    wifiNetworks?: IWifiSummary[] | undefined;
};
const WifiSummaryTable: React.FC<WifiSummaryTableProps> = (props: WifiSummaryTableProps): React.ReactElement => {
    const getWifiSummary = (ssid: string, strength: number, id: number) => {
        const selectedWifi = (ssid: string) => {
            console.log('ssid called', ssid);
        };
        return (
            <DeviceField
                label={ssid}
                value={strength.toString()}
                key={`df-${id}-${ssid}-${strength}`}
                clickHandler={() => {
                    selectedWifi(ssid);
                }}
            />
        );
    };

    const getSummaryTable = (wifiNetworks: IWifiSummary[] | undefined) => {
        if (typeof wifiNetworks === 'undefined') {
            return;
        }

        const summaryNetworks = wifiNetworks.map((wifi, i) => {
            return getWifiSummary(wifi.ssid, wifi.signalStrength, i);
        });

        return summaryNetworks;
    };
    return (
        <IonGrid>
            <IonRow>
                <IonCol>
                    <IonText>{props.ssidLabel}</IonText>
                </IonCol>
                <IonCol>
                    <IonText>{props.signalStength}</IonText>
                </IonCol>
            </IonRow>
            {getSummaryTable(props.wifiNetworks)}
        </IonGrid>
    );
};

export default WifiSummaryTable;
