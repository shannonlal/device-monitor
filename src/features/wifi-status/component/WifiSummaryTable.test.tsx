import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import WifiSummaryTable from './WifiSummaryTable';
import { IWifiSummary } from '../../../interfaces/models';

describe('WifiSummaryTable', () => {
    test('rendering a simple wifi summary table', async () => {
        const ssidLabel = 'Network Name';
        const signalStrengthLabel = 'Signal Strength';
        const mockOnClick = jest.fn();
        const mockWifSummary: IWifiSummary[] = [
            {
                ssid: 'Test Wifi 1',
                signalStrength: 3,
            },
            {
                ssid: 'Test Wifi 4',
                signalStrength: 2,
            },
        ];

        const { asFragment } = render(
            <WifiSummaryTable
                ssidLabel={ssidLabel}
                signalStength={signalStrengthLabel}
                wifiNetworks={mockWifSummary}
                selectWifi={mockOnClick}
            />,
        );

        expect(asFragment()).toMatchSnapshot();
    });

    test('rendering a wifi summary table without wifi summary', async () => {
        const ssidLabel = 'Network Name';
        const signalStrengthLabel = 'Signal Strength';
        const mockOnClick = jest.fn();

        const { asFragment } = render(
            <WifiSummaryTable ssidLabel={ssidLabel} signalStength={signalStrengthLabel} selectWifi={mockOnClick} />,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    test('rendering a simple wifi summary table and click on it', async () => {
        const ssidLabel = 'Network Name';
        const signalStrengthLabel = 'Signal Strength';
        const mockOnClick = jest.fn();
        const mockWifSummary: IWifiSummary[] = [
            {
                ssid: 'Test Wifi 1',
                signalStrength: 3,
            },
            {
                ssid: 'Test Wifi 4',
                signalStrength: 2,
            },
        ];

        const { asFragment, getByText } = render(
            <WifiSummaryTable
                ssidLabel={ssidLabel}
                signalStength={signalStrengthLabel}
                wifiNetworks={mockWifSummary}
                selectWifi={mockOnClick}
            />,
        );

        const field = getByText('Test Wifi 1');
        console.log('test',field.firstChild);
        fireEvent.click(field.firstChild);


        expect(asFragment()).toMatchSnapshot();
    });
});
