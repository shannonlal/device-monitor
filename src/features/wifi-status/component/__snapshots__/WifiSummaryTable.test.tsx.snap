// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WifiSummaryTable rendering a simple wifi summary table 1`] = `
<DocumentFragment>
  <ion-grid>
    <ion-row>
      <ion-col>
        <ion-text>
          Network Name
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          Signal Strength
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          Test Wifi 1
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          3
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          Test Wifi 4
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          2
        </ion-text>
      </ion-col>
    </ion-row>
  </ion-grid>
</DocumentFragment>
`;

exports[`WifiSummaryTable rendering a simple wifi summary table and click on it 1`] = `
<DocumentFragment>
  <ion-grid>
    <ion-row>
      <ion-col>
        <ion-text>
          Network Name
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          Signal Strength
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          Test Wifi 1
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          3
        </ion-text>
      </ion-col>
    </ion-row>
    <ion-row>
      <ion-col>
        <ion-text>
          Test Wifi 4
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          2
        </ion-text>
      </ion-col>
    </ion-row>
  </ion-grid>
</DocumentFragment>
`;

exports[`WifiSummaryTable rendering a wifi summary table without wifi summary 1`] = `
<DocumentFragment>
  <ion-grid>
    <ion-row>
      <ion-col>
        <ion-text>
          Network Name
        </ion-text>
      </ion-col>
      <ion-col>
        <ion-text>
          Signal Strength
        </ion-text>
      </ion-col>
    </ion-row>
  </ion-grid>
</DocumentFragment>
`;
